windows/ReactNativeWebView/ReactWebViewManager.cpp

```diff
@@ -0,0 +1,173 @@
+                        if (srcMap.find("__packager_asset") != srcMap.end()) {
+                            isPackagerAsset = srcMap.at("__packager_asset").Boolean();
+                        }
+
+                        if (isPackagerAsset && uriString.find("assets") == 0) {
+                            uriString.replace(0, 6, "ms-appx://");
+                        }
+
+                        SetSource(winrt::Uri(to_hstring(uriString)));
+                    }
+                }
+            }
+        }
+    }
+
+    void ReactWebViewManager::SetSource(winrt::Uri const& uri) {
+        if (auto webView = m_webView.get()) {
+            auto tag = webView.GetValue(winrt::FrameworkElement::TagProperty()).as<winrt::IPropertyValue>().GetInt64();
+            m_reactContext.DispatchEvent(
+                webView,
+                L"topLoadingStart",
+                [webView, tag, uri](winrt::IJSValueWriter const& eventDataWriter) noexcept {
+                    eventDataWriter.WriteObjectBegin();
+                    {
+                        WriteProperty(eventDataWriter, L"canGoBack", webView.CanGoBack());
+                        WriteProperty(eventDataWriter, L"canGoForward", webView.CanGoForward());
+                        WriteProperty(eventDataWriter, L"loading", true);
+                        WriteProperty(eventDataWriter, L"target", tag);
+                        WriteProperty(eventDataWriter, L"title", webView.DocumentTitle());
+                        WriteProperty(eventDataWriter, L"url", uri.ToString());
+                    }
+                    eventDataWriter.WriteObjectEnd();
+                });
+
+            webView.Navigate(uri);
+        }
+    }
+
+    // IViewManagerWithExportedEventTypeConstants
+    ConstantProvider ReactWebViewManager::ExportedCustomBubblingEventTypeConstants() noexcept {
+        return nullptr;
+    }
+
+    ConstantProvider ReactWebViewManager::ExportedCustomDirectEventTypeConstants() noexcept {
+        return [](winrt::IJSValueWriter const& constantWriter) {
+            WriteCustomDirectEventTypeConstant(constantWriter, "onLoadingStart");
+            WriteCustomDirectEventTypeConstant(constantWriter, "onLoad");
+            WriteCustomDirectEventTypeConstant(constantWriter, "onLoadingFinish");
+        };
+    }
+
+    // IViewManagerWithCommands
+    IMapView<hstring, int64_t> ReactWebViewManager::Commands() noexcept {
+        auto commands = winrt::single_threaded_map<hstring, int64_t>();
+        commands.Insert(L"goForward", 0);
+        commands.Insert(L"goBack", 1);
+        commands.Insert(L"reload", 2);
+        return commands.GetView();
+    }
+
+    void ReactWebViewManager::DispatchCommand(
+        FrameworkElement const& view,
+        int64_t commandId,
+        winrt::IJSValueReader const& commandArgsReader) noexcept {
+        if (auto webView = view.try_as<winrt::WebView>()) {
+            switch (commandId) {
+            case 0: // goForward

```

Comment:
It would be nice if these were named constants.

Reply:
Updated.